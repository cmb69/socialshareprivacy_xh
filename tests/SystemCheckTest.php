<?php

namespace Socialshareprivacy;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Plib\FakeSystemChecker;
use Plib\SystemChecker;

class SystemCheckTest extends TestCase
{
    private SystemChecker $systemChecker;

    protected function setUp(): void
    {
        global $plugin_tx;
        $plugin_tx = XH_includeVar("./languages/en.php", "plugin_tx");
        $this->systemChecker = new FakeSystemChecker();
    }

    private function sut(): SystemCheck
    {
        return new SystemCheck("./plugins/socialshareprivacy", $this->systemChecker);
    }

    public function testRendersSystemCheck(): void
    {
        $response = $this->sut()->render();
        Approvals::verifyHtml($response);
    }
}
