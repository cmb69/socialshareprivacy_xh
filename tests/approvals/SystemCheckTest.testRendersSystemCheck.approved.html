<h4>System check</h4>
<img src="./plugins/socialshareprivacyimages/fail.png" alt="Failure">&nbsp;&nbsp;PHP version &ge; 7.4.0<br>
<img src="./plugins/socialshareprivacyimages/fail.png" alt="Failure">&nbsp;&nbsp;CMSimple_XH version &ge; 1.7.0<br><br>
<img src="./plugins/socialshareprivacyimages/warn.png" alt="Warning">&nbsp;&nbsp;Folder './plugins/socialshareprivacyconfig/' writable<br>
<img src="./plugins/socialshareprivacyimages/warn.png" alt="Warning">&nbsp;&nbsp;Folder './plugins/socialshareprivacycss/' writable<br>
<img src="./plugins/socialshareprivacyimages/warn.png" alt="Warning">&nbsp;&nbsp;Folder './plugins/socialshareprivacylanguages/' writable<br>
